### PR TITLE
holybro can gps v1 power - with nuttx deglich

### PR DESCRIPTION
**Pull request by davids5**
~~This just a test. First commit must come in, but second is a WIP~~

This is ready to come in. It will bring in the GPIO deglitch backports and USB unstuck from NuttX.

The same class of error: Where the I2C init is called in px4_paltform_init without the sensor power ON can happen on other boards as well. We should have a look at that.

### CODE REVIEW

**Comment by davids5** [comment]
@dagar - this needs to run on the test rack

**Comment by davids5** [comment]
@AlexKlimaj - Please retest this I have fully deglitched the GPIO code.

**Comment by AlexKlimaj** [comment]
This appears to solve https://github.com/PX4/PX4-Autopilot/issues/18237

Scope shows the glitches are gone.

Power comes on, bus gets pulled up.
image

Bus reset is called
image

The init with the 00 write
image

Then the driver starts
image

**Review by AlexKlimaj — DISMISSED** [review]

**Comment by davids5** [comment]
Thans @AlexKlimaj for cross testing. I need to see how the test rack runs with this before unstreaming the NuttX changes and back porting.

File: boards/holybro/can-gps-v1/src/init.c [hunk 1]
```diff
@@ -135,6 +135,8 @@ stm32_boardinitialize(void)
 
 __EXPORT int board_app_initialize(uintptr_t arg)
 {
+	VDD_3V3_SENSORS_EN(true);
```
Comment (dagar) [inline boards/holybro/can-gps-v1/src/init.c:138]: If you're going to do it manually should PC15 be taken away from SPI bus power control?

https://github.com/PX4/PX4-Autopilot/blob/8d46772444e19a7b29534f435b97187390221681/boards/holybro/can-gps-v1/src/spi.cpp#L41

**Comment by mrpollo** [comment]
I have a CAN GPS from Holybro, I'm up for testing if you need anything else @davids5

File: boards/holybro/can-gps-v1/src/init.c (hunk 1 above)
Comment (davids5) [reply boards/holybro/can-gps-v1/src/init.c:138]: It is a good point. 

Initial state is Off.
VDD_3V3_SENSORS_EN(true); sets it on
stm32_spiinitialize will turn it off and then on again.

image

But it us not needed. The time to move the rail from 3.3V to 3.13V (-5%) is about 100 uS. 

The time the pin is then turned off and on again in stm32_spiinitialize  is about 2.2 uS. The rail will not move.

**Comment by davids5** [comment]
@dagar this will bring in the GPIO deglitch backports and usb unstuck from NuttX.

**Review by dagar — APPROVED** [review]

**Pull request merged**